src/lib.rs

```diff
@@ -88,42 +90,167 @@ impl InlinerBuilder {
         };
         let result =
             Visitor::<R>::with_resolver(src_file, self.root, errors.as_mut(), Cow::Owned(resolver))
-                .visit();
+                .visit()
+                .map_err(|kind| Error::Initial(kind))?;
         match errors {
             Some(ref errors) if errors.is_empty() => Ok(result),
             None => Ok(result),
-            Some(errors) => Err(Error::NotFound(errors)),
+            Some(errors) => Err(Error::Inline(errors)),
         }
     }
 }
 
-/// A source location
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct SourceLocation {
-    pub path: PathBuf,
-    pub line: usize,
-    pub column: usize,
-    _private: (),
+/// An error that was encountered while reading, parsing or inlining a module.
+#[derive(Debug)]
+#[non_exhaustive]
```

Comment:
If we omit the use of this attribute, should this otherwise work on 1.32.0? Despite having written multiple proc-macro crates, I've never really gotten my head around how rustc versions and proc-macros interact, so I'm not sure if bumping the min-version of this crate would lead to its dependent crates also having to bump their rustc version.

If removing this attribute lets us keep 1.32.0 at the expense of requiring a minor version bump for the next error case that comes along, I think that'd be an acceptable tradeoff. I think you've been thoughtful enough in covering the cases that I don't think it's likely we'll need more any time soon.